File: app/src/main/java/com/example/redscate/ChatActivity.java
package com.example.redscate;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.view.LayoutInflater;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;

public class ChatActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_chat);

        LinearLayout messagesContainer = findViewById(R.id.messagesContainer);

        findViewById(R.id.chipHola).setOnClickListener(v ->
                addMessage(messagesContainer, getString(R.string.hola)));
        findViewById(R.id.chipNeed).setOnClickListener(v ->
                addMessage(messagesContainer, getString(R.string.chip_need)));
        findViewById(R.id.chipEmergency).setOnClickListener(v ->
                startActivity(new Intent(this, ChatAlertActivity.class)));
        findViewById(R.id.safeButton).setOnClickListener(v ->
                startActivity(new Intent(this, ChatAlertActivity.class)));
    }

    private void addMessage(LinearLayout messagesContainer, String text) {
        View messageBlock = LayoutInflater.from(this)
                .inflate(R.layout.item_chat_message_self, messagesContainer, false);
        TextView messageText = messageBlock.findViewById(R.id.messageText);
        TextView messageTimeText = messageBlock.findViewById(R.id.messageTimeText);

        messageText.setText(text);
        String timestamp = new SimpleDateFormat("HH:mm:ss dd/MM/yyyy", Locale.getDefault()).format(new Date());
        messageTimeText.setText(getString(R.string.send_time_prefix) + " " + timestamp);
        messagesContainer.addView(messageBlock);
    }
}

File: app/src/main/java/com/example/redscate/ChatActivity.kt
package com.example.redscate

import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.widget.LinearLayout
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

class ChatActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_chat)

        val messagesContainer: LinearLayout = findViewById(R.id.messagesContainer)

        findViewById<View>(R.id.chipHola).setOnClickListener {
            addMessage(messagesContainer, getString(R.string.hola))
        }
        findViewById<View>(R.id.chipNeed).setOnClickListener {
            addMessage(messagesContainer, getString(R.string.chip_need))
        }
        findViewById<View>(R.id.chipEmergency).setOnClickListener {
            startActivity(Intent(this, ChatAlertActivity::class.java))
        }
        findViewById<View>(R.id.safeButton).setOnClickListener {
            startActivity(Intent(this, ChatAlertActivity::class.java))
        }
    }

    private fun addMessage(messagesContainer: LinearLayout, text: String) {
        val messageBlock = LayoutInflater.from(this)
            .inflate(R.layout.item_chat_message_self, messagesContainer, false)
        val messageText: TextView = messageBlock.findViewById(R.id.messageText)
        val messageTimeText: TextView = messageBlock.findViewById(R.id.messageTimeText)

        messageText.text = text
        val timestamp = SimpleDateFormat("HH:mm:ss dd/MM/yyyy", Locale.getDefault()).format(Date())
        messageTimeText.text = "${getString(R.string.send_time_prefix)} $timestamp"
        messagesContainer.addView(messageBlock)
    }
}
